### PR TITLE
wip: add p2pk tests

### LINKED ISSUE #183: [feat] P2PK transaction support
Currently, all the opcodes for P2PK are implemented. To ensure we support all P2PK transactions/formats, we will need to create a set of tests for these transaction types and ensure all the logic and edge cases are handled properly. Also, including a P2PK transaction from mainnet as a test would be ideal.

#### References
[learn P2PK & examples](https://learnmeabitcoin.com/technical/script/p2pk/)
[example crypto tests](https://github.com/keep-starknet-strange/shinigami/blob/main/src/opcodes/tests/test_crypto.cairo)

### PR DESCRIPTION
**Pull request by bloomingpeach**
- [ ] Fixes #183 
- [ ] follows contribution [guide](https://github.com/keep-starknet-strange/shinigami/blob/main/CONTRIBUTING.md)
- [ ] code change includes tests

### CODE REVIEW

File: src/signature/utils.cairo [hunk 1]
```diff
@@ -8,12 +8,20 @@ pub fn remove_opcodeseparator(script: @ByteArray) -> @ByteArray {
...
+            continue;
+        }
+        let data_len = Opcode::data_len(i, script).unwrap();
+        let end = i + data_len + 1;
+        while i < end {
+            parsed_script.append_byte(script[i]);
+            i += 1;
         }
-        i += 1;
     };
```
Comment (b-j-roberts) [inline src/signature/utils.cairo:25]: @bloomingpeach Looks like there was a bug in this function that caused the transaction to be parsed incorrectly when computing the sighash. After fixing it, you test is passing!

**Review by b-j-roberts — APPROVED** [review]
I'll go ahead and merge now, thank you for looking into this and building the tests!
